<a href="https://colab.research.google.com/github/Annamalag/PI_Mineria_Datos_1-/blob/main/Notebooks/01_inspeccion_inicial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

# Proyecto Integrador - Minería de Datos

## Notebook 01 - Inspección Inicial del Dataset

### Objetivo

Realizar una exploración inicial del dataset para conocer su estructura, identificar tipos de datos, valores faltantes, registros duplicados y obtener estadísticas descriptivas antes del proceso de limpieza.

# **Cargar el dataset**

In [ ]:
df = pd.read_json('dataset.json')

In [ ]:
df.head(10)

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1
5,10005,20,Básico,670.2,Uruguay,Drama,2020-07-03,2
6,10006,37,Básico,346.6,Perú,Thriller,2019-07-26,1
7,10007,31,Estándar,974.6,Chile,Acción,2019-02-24,1
8,10008,36,Premium,1432.2,Colombia,Romance,2025-08-03,2
9,10009,37,Estándar,1375.4,Argentina,Thriller,2024-02-12,1


# **Información general**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   object 
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   object 
 5   favorite_genre            7920 non-null   object 
 6   last_login_date           7840 non-null   object 
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 510.1+ KB


In [ ]:
print("Dimensiones:", df.shape)

Dimensiones: (8160, 8)


# **Tipos de datos**

In [ ]:
df.dtypes

,0
user_id,int64
age,int64
subscription_plan,object
monthly_watch_time_mins,float64
country,object
favorite_genre,object
last_login_date,object
customer_support_tickets,int64


# **Estadísticos**

In [ ]:
df.describe().round(2)

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.00,8160.00,7967.00,8160.00
mean,13995.43,34.10,1107.35,1.80
std,2310.81,14.51,5310.44,11.33
min,10000.00,-5.00,-120.00,-1.00
25%,11987.75,25.00,489.20,0.00
50%,13998.50,33.00,757.40,1.00
75%,15997.25,42.00,1045.70,1.00
max,17999.00,150.00,99999.00,150.00


In [ ]:
df.describe(include="object")

,subscription_plan,country,favorite_genre,last_login_date
count,8160,8160,7920,7840
unique,15,26,28,3062
top,Básico,Brasil,Comedia,2026-15-03
freq,3450,1132,1112,20


# **Valores faltantes**

In [ ]:
df.isnull().sum()

,0
user_id,0
age,0
subscription_plan,0
monthly_watch_time_mins,193
country,0
favorite_genre,240
last_login_date,320
customer_support_tickets,0


# **Registros duplicados**

In [ ]:
df.duplicated().sum()

np.int64(126)

# **Valores únicos**

In [ ]:
for columna in ["subscription_plan","country","favorite_genre"]:
    print("="*50)
    print(columna)
    print(df[columna].unique())

subscription_plan
['Estándar' 'Básico' 'Premium' 'Std' 'estandar' 'basico' 'básico'
 'Premium ' 'premium' 'Premiun' 'BASICO' 'STANDARD' 'Basic' 'Estándar '
 'PREMIUM']
country
['Brasil' 'Colombia' 'Uruguay' 'Perú' 'Chile' 'Argentina' 'México'
 'Brazil' 'brasil' 'méxico' 'chile' 'uruguay' 'MEX' 'ARG' 'colombia' 'COL'
 'Mexico' 'URY' 'Chile ' 'Peru' 'perú' 'argentina' 'PER' 'Argentina '
 'BRA' 'CHL']
favorite_genre
['Crime' 'Thriller' 'Drama' 'Acción' 'Romance' 'Comedia' 'Documental'
 'ACCIÓN' None 'CRIME' 'Comedia ' 'comedy' 'DRAMA' 'THRILLER'
 'Documentary' 'Thriller ' 'accion' 'Action' 'COMEDIA' 'ROMANCE' 'thriler'
 'DOC' 'romance' 'Crimen' 'Drama ' 'drama' 'Romance ' 'documental' 'crime']


### **Conclusiones de la inspección inicial**

En la exploración inicial,
Se observo:

El dataset contiene 8.160 registros y 8 variables.

Existen 126 filas duplicadas

Se detectan valores faltantes en:


*   monthly_watch_time_mins(193)
*   favorite_genre(240)
*   last_login_date(320)


La variable last_login_date se encuentra como tipo object, por lo que será necesario convertirla a formato fecha (datetime).

Las variables subscription_plan, country y favorite_genre son categóricas.
A simple vista se observan posibles problemas de calidad que deberán tratarse en la siguiente etapa, por ejemplo:
Diferentes formas de escribir un mismo país (Brasil, Brazil, brasil).
Diferentes formas de escribir un mismo plan (Estándar, estandar, Std, Premium , etc.).
Diferentes escrituras de géneros (Acción, ACCIÓN, accion, DRAMA, Thriller , etc.).

Fechas con distintos formatos e incluso algunas aparentemente inválidas.

Posibles valores atípicos en age y monthly_watch_time_mins (por ejemplo edades de -5 o 150 años, y tiempos de visualización extremadamente altos).
